commands/operator-sdk/cmd/scorecard/basic_tests.go

```diff
@@ -0,0 +1,186 @@
+
+package scorecard
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"strings"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// checkSpecAndStat checks that the spec and status blocks exist. If noStore is set to true, this function
+// will not store the result of the test in sCTests and will instead just wait wait until the spec and
+// status blocks exist or return an error after the timeout.
+func checkSpecAndStat(runtimeClient client.Client, obj unstructured.Unstructured, noStore bool) error {
+	testSpec := scorecardTest{testType: basicOperator, name: "Spec Block Exists", maximumPoints: 1}
+	testStat := scorecardTest{testType: basicOperator, name: "Status Block Exist", maximumPoints: 1}
+	var specPoints, statusPoints int
+	err := wait.Poll(time.Second*1, time.Second*time.Duration(SCConf.InitTimeout), func() (bool, error) {
+		err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+		if err != nil {
+			return false, fmt.Errorf("error getting custom resource: %v", err)
+		}
+		pass := true
+		if obj.Object["spec"] == nil {
+			pass = false
+			specPoints = 0
+		} else {
+			specPoints = 1
+		}
+		if obj.Object["status"] == nil {
+			pass = false
+			statusPoints = 0
+		} else {
+			statusPoints = 1
+		}
+		return pass, nil
+	})
+	if !noStore {
+		testSpec.earnedPoints = specPoints
+		testStat.earnedPoints = statusPoints
+		scTests = append(scTests, testSpec)
+		scTests = append(scTests, testStat)
+	}
+	if err != nil && !reflect.DeepEqual(err, wait.ErrWaitTimeout) {
+		return err
+	}
+	return nil
+}
+
+// TODO: user specified tests for operators
+
+// checkStatusUpdate looks at all fields in the spec section of a custom resource and attempts to modify them and
+// see if the status changes as a result. This is a bit prone to breakage as this is a black box test and we don't
+// know much about how the operators we are testing actually work and may pass an invalid value. In the future, we
+// should use user-specified tests
+func checkStatusUpdate(runtimeClient client.Client, obj unstructured.Unstructured) error {
+	test := scorecardTest{testType: basicOperator, name: "Operator actions are reflected in status", maximumPoints: 1}
+	err := runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: SCConf.Namespace, Name: name}, &obj)
+	if err != nil {
+		return fmt.Errorf("error getting custom resource: %v", err)
+	}
+	if obj.Object["status"] == nil || obj.Object["spec"] == nil {
+		scTests = append(scTests, test)
+		return nil
+	}
+	statCopy := make(map[string]interface{})
+	for k, v := range obj.Object["status"].(map[string]interface{}) {
+		statCopy[k] = v
+	}
+	pass := true
```

Comment:
nit: can we move this closer to when it is used?

Also, why is the default that the test is passing?